tests/test_0662-rntuple-stl-containers.py

```diff
@@ -0,0 +1,59 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/uproot4/blob/main/LICENSE
+
+import json
+import queue
+import sys
+
+import awkward as ak
+import numpy
+import pytest
+import skhep_testdata
+
+import uproot
+
+
+def test_rntuple_stl_containers():
+    filename = skhep_testdata.data_path("test_ntuple_stl_containers.root")
+    with uproot.open(filename) as f:
+        R = f["ntuple"]
+        assert R.keys() == [
+            "string",
+            "vector_int32",
+            "vector_vector_int32",
+            "vector_string",
+            "vector_vector_string",
+            "variant_int32_float",
+            "vector_variant_int32_float",
+            "tuple_int32_string",
+            "vector_tuple_int32_string",
+        ]
+        r = R.arrays()
+        assert ak.all(r["string"] == ["one", "two"])
+
+        assert r["vector_int32"][0] == [1]
+        assert ak.all(r["vector_int32"][1] == [1, 2])
+        assert r["vector_vector_int32"][0] == [[1]]
+
+        assert ak.all(r["vector_vector_int32"][1] == [[1], [2]])
+
+        assert r["vector_string"][0] == ["one"]
+        assert ak.all(r["vector_string"][1] == ["one", "two"])
+
+        assert ak.all(r["vector_vector_string"] == [["one"], [["one"], ["two"]]])
+        assert type(r["variant_int32_float"][0]) == numpy.float32
+        assert type(r["variant_int32_float"][1]) == numpy.float32
+
+        assert type(r["vector_variant_int32_float"][0][0]) == numpy.int32
+        assert type(r["vector_variant_int32_float"][1][0]) == numpy.int32
+        assert type(r["vector_variant_int32_float"][1][1]) == numpy.float32
```

Comment:
Integers and floating point numbers are close enough that this test (checking `type`) could legitimately break in the future, if Awkward takes a new policy on return types. (This, for instance, would fail in Awkward v1. It's recognized as a backward-incompatible change, of course.) Awkward's `simplify_uniontype` would turn a union of integers and floating point numbers into an array of floating point numbers, following NumPy's type promotion rules, and maybe that would get used on this someday.

A better test of variant would be integers and strings: those are very different from one another and would never be merged upon `simplify_uniontype`.

Reply:
let me make a new stl container test file and make a PR to the test file repo first...

Reply:
addressed in https://github.com/scikit-hep/scikit-hep-testdata/pull/82